### PR TITLE
Gameplay & UI intergration

### LINKED ISSUE #14: Draw ui over the map (add styrebrett)
Interaction with [this ](https://raw.githubusercontent.com/wiki/inf112-v19/Blodige-roboter/02-02-2019.jpg)drawing is also included in this issue

Utvikler: @kh498 
Tester: @rikkeaas

### PR DESCRIPTION
**Pull request by kh498**
Fixes #14

### CODE REVIEW

**Comment by DanielBerge** [comment]
Any testing done by @rikkeaas ?

**Comment by kh498** [comment]
Fixed all bugs found, done with testing @rikkeaas?

**Comment by rikkeaas** [comment]
Done with tests except the "Test all control panel event listeners...". I didn't really understand what was happening there so I don't know how to test it. So I need some help with that :) 

I think the rest can be reviewed now though and we can add the last tests afterwards? @kh498 @DanielBerge

File: src/main/java/no/uib/inf112/core/ui/UIHandler.java [hunk 1]
```diff
@@ -2,44 +2,211 @@
...
         stage.act(Gdx.graphics.getDeltaTime()); //Perform ui logic
         stage.draw(); //Draw the ui
     }
 
-    public Stage getStage() {
-        return stage;
+    public void resize() {
+        stage.getViewport().update(Gdx.graphics.getWidth(), Gdx.graphics.getHeight(), true);
+
+        //make sure we have some background around the down down button
```
Comment (kapir) [inline src/main/java/no/uib/inf112/core/ui/UIHandler.java]: down down button? what does this mean?

Comment (kapir) [reply src/main/java/no/uib/inf112/core/ui/UIHandler.java]: UI?

File: src/main/java/no/uib/inf112/core/player/Player.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+                robot.move(cards[event.getId()].getAction());
+            });
+        }
+    }
+
+    public void doTurn() {
+        //TODO Issue #44 check if dead
+        //TODO Issue #24 check if is powered down (then heal)
+        for (Card card : cards) {
+            robot.move(card.getAction());
```
[the remark this answers is not in the document]
Comment (kh498) [reply src/main/java/no/uib/inf112/core/player/Player.java]: This is temp code to be moved to PlayerHandler, (same with deck in player)

File: src/main/java/no/uib/inf112/core/player/Direction.java [hunk 3]
```diff
@@ -5,10 +5,73 @@
...
+    private final int dx;
+    private final int dy;
+
+    Direction(int dx, int dy) {
+
+        this.dx = dx;
+        this.dy = dy;
+    }
+
+    public int getDx() {
```
[the remark this answers is not in the document]
Comment (kh498) [reply src/main/java/no/uib/inf112/core/player/Direction.java]: In my experience dx/dy is a very common abbreviation for delta x/y. So I would say it does not really matter which one we choose. However we should be consistent on which one we are using in the project. 

(sidenote: Robot.move(int, int) uses `deltaX` while everything else is using `dx`)

File: src/main/java/no/uib/inf112/core/player/Direction.java [hunk 4]
```diff
@@ -5,10 +5,73 @@
...
+        return dx;
+    }
+
+    public int getDy() {
+        return dy;
+    }
+
+    public Direction inverse() {
```
[the remark this answers is not in the document]
Comment (kh498) [reply src/main/java/no/uib/inf112/core/player/Direction.java]: Good spot

File: src/main/java/no/uib/inf112/core/player/Player.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (kapir) [reply src/main/java/no/uib/inf112/core/player/Player.java]: Ok, just comments say excactly that, also maybe a #33 reference aswell

File: src/main/java/no/uib/inf112/core/player/Direction.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (kapir) [reply src/main/java/no/uib/inf112/core/player/Direction.java]: Then lets have dx for now, we can discuss this in the next meeting to make sure everyone is familar with the abbreviation, i'll add it to the agenda

File: src/main/java/no/uib/inf112/core/player/Player.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (kh498) [reply src/main/java/no/uib/inf112/core/player/Player.java]: That being said, maybe we should inform @DanielBerge about this

File: src/main/java/no/uib/inf112/core/ui/UIHandler.java (hunk 1 above)
Comment (kh498) [reply src/main/java/no/uib/inf112/core/ui/UIHandler.java]: power down button 🥇

**Pull request merged**